gospel/pps/gospel_pps.mll

```diff
@@ -0,0 +1,68 @@
+{
+  type t = Gospel of string | Other of string
+
+  let queue = Queue.create ()
+  let empty_line = ref true
+  let buf = Buffer.create 1024
+
+  let push () =
+    if Buffer.length buf > 0 then (
+      Queue.push (Other (Buffer.contents buf)) queue;
+      Buffer.clear buf)
+
+  let flush () =
+    push ();
+    let print = function
+      | Gospel s ->
+          let ats = if !empty_line then "@@@" else "@@" in
+          Format.printf "[%sgospel \"%s\"]" ats s
```

Comment:
If I understand it correctly, we assume we put GOSPEL comments always in places where they can be indeed turned into @@ or @@@ attributes. Correct? 

I think for signature files this is perfectly fine. We will run into some problems if you start putting GOSPEL in the middle of code, e.g.,:
```
  while b do 
    (*@ invariant I *)
    e
  done
```
The GOSPEL comment would be turned into a @@ attribute, which is refused by the OCaml parser.

Yet again, I think this perfectly fine for signature files. We will have plenty of time do discuss such matters for implementations.

Reply:
Ah indeed, this only works for signatures at the moment, and we'll have to figure out something a bit different for implementations.